feature/amazon/src/main/java/com/revenuecat/purchases/amazon/purchasing/ProxyAmazonBillingDelegate.kt

```diff
@@ -0,0 +1,64 @@
+package com.revenuecat.purchases.amazon.purchasing
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.os.ResultReceiver
+import androidx.annotation.VisibleForTesting
+import com.amazon.device.iap.model.RequestId
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.amazon.AmazonStrings
+import com.revenuecat.purchases.amazon.ProxyAmazonBillingActivity
+import com.revenuecat.purchases.amazon.ProxyAmazonBillingActivityBroadcastReceiver
+import com.revenuecat.purchases.amazon.PurchasingServiceProvider
+import com.revenuecat.purchases.common.errorLog
+
+internal class ProxyAmazonBillingDelegate {
+
+    @JvmSynthetic
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var broadcastReceiver: ProxyAmazonBillingActivityBroadcastReceiver? = null
+    private val filter = ProxyAmazonBillingActivityBroadcastReceiver.newPurchaseFinishedIntentFilter()
+
+    fun onCreate(activity: Activity, savedInstanceState: Bundle?) {
+        broadcastReceiver = ProxyAmazonBillingActivityBroadcastReceiver(activity)
+        activity.applicationContext.registerReceiver(broadcastReceiver, filter)
```

Comment:
Hmm I noticed we are now registering the receiver in the application context, instead of the `ProxyAmazonBillingActivity`. I think it would be safer on the activity, since if we leave it on the application, and the `onDestroy` doesn't get called for any reason, the application can remain registered forever.